sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -756,7 +621,9 @@ public void populateDisplayData(DisplayData.Builder builder) {
       for (Map.Entry<String, Object> conf : getConsumerConfig().entrySet()) {
         String key = conf.getKey();
         if (!ignoredConsumerPropertiesKeys.contains(key)) {
-          builder.add(DisplayData.item(key, ValueProvider.StaticValueProvider.of(conf.getValue())));
+          Object value = DisplayData.inferType(conf.getValue()) != null
```

Comment:
The issue is with DisplayData. It throws if inferType() does not succeed, i.e. it only handles a fixed set of types, but consumer config values could be any (serializable) objects. For such objects, we use toString() to display.